### DISCUSSION TITLE
Milestones cleanup

### DISCUSSION THREAD

**Discussion by YBAZAN** [AvaloniaUI/Avalonia#6886] [General]
Avalonia looks to be great UI framework now. But I find difficult to project with the different milestones not completely completed.

As far as I understand, the active version is the 0.10. But according to the milestones, some features / issues are stil not closed in 0.9. 

What is about 1.0 and Post-1.0 versions (older than 0.11 one) ?

I feel a cleanup would be welcome. And a clearer roadmap as well...

**Comment by robloo** [comment]
Definitely agree, any such cleanup also needs to go through the substantial backlog of PRs. It's been a few years since that was cleaned out.

Maybe this is something where each core developer and then some volunteers could take N-number of PRs so it doesn't all fall onto 1 or 2 people.

**Reply by YBAZAN** [reply]
I'm a little bit surprised about the lack of feedback on this topic.

It's good to have news ideas, enrich the library, but the lack of vision about it will make it difficult to promote.

That's only my point of view, it's still a good library...

**Reply by robloo** [reply]
I share some of the concerns. A few points:

 1) A lot of responsibility is on those that filed the PRs to begin with. They need to push them closed as well.
 2) This is very much an open source project that is primarily volunteer driven. That means people only work on what they want to work on when they feel like it.
 3) Several of the core maintainers have formed a company for paid support. That's great and they deserve income for everything they have done. That also means they don't seem to contribute as much anymore unless their is a paying sponsor. Perhaps this helps prioritize the most important and widely useful features for real apps as well.
 4) For those of us on the outside there seems to be no direction or central leadership to the project. That might be more of a communication issue. I know the core team has frequent meetings; however, the minutes from those meetings are never posted so we don't know what was discussed or what they were planning next. Along with point 3, a lot of the planning seems to be driven by paying customers and business needs as well.

**Reply by maxkatz6** [reply]
FYI milestones are being cleanup for a while now. Using github private projects though as an experiment (new GH feature). When it will agreed, it can go public.

**Reply by jmacato** [reply]
@robloo fwiw we dont have closed door meetings. we constantly coordinate over group chats instead
that means that we don't have any minutes to share.. 

nevertheless this seems to be a communications issue that we need to address.

**Comment by grokys** [comment]
Yes, it's true that our milestones issue is rather out of date. As you probably know, priorities change, and we've not been keeping the milestones up-to-date. However we're currently in planning for the next release and are in the process of tidying up these milestones. As @jmacato mentioned we do this all in a chat room which isn't open to the public - back in the day we used to do it all in the public gitter room but as Avalonia got more popular it became impossible to discuss such things in public due to the cross-talk with people asking general questions. We could perhaps open up our chat read-only but I'm not sure how useful this would be, and I'd be afraid that the random ideas we often talk about would be misinterpreted as solid plans.

To address some points:

>Several of the core maintainers have formed a company for paid support. That's great and they deserve income for everything they have done. That also means they don't seem to contribute as much anymore unless their is a paying sponsor. Perhaps this helps prioritize the most important and widely useful features for real apps as well.

Yes, unfortunately this is true. In the last year Avalonia has really taken off, but we're still a small company and we have to prioritize paid customer's requests. This means that we get less time to address issues that may be important to the community but haven't affected our paid customers. _This isn't good, and we don't want it to be this way._ However we plan to (and have started to) reinvest the income by paying people to work on Avalonia, which I'm sure you'll agree is a better place to be than relying on solely volunteers.

> For those of us on the outside there seems to be no direction or central leadership to the project. That might be more of a communication issue. 

This is both true and not true IMO. As I said, we working on updating a plan for the next release, but as I've also said we're also a small company and we don't have product managers etc. In addition, until a year an a half ago we were purely volunteers and you can't tell volunteers what to work on ;) We still retain that mindset now: people are free to work on what they want to work on (outside paying customer support). This has upsides and downsides and I think as we grow you'll see a bit more formalism coming, but we're still young and trying to find the right balance.

> Along with point 3, a lot of the planning seems to be driven by paying customers and business needs as well.

Actually this one is less true. Most of our customer's needs right come come into the short-term bugfix category. We obviously see where they're having problems but most of the time we're aware of/had experienced these pain points ourselves and had already discussed addressing them.

**Comment by robloo** [comment]
The Project board has been updated a bit for the next release [11.0](https://github.com/AvaloniaUI/Avalonia/projects/10).

The backlog of PRs is growing and it would be great for clarity on how to close these out. The community can help, just need some direction.
